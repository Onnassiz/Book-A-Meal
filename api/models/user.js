
module.exports = (sequelize, DataTypes) => {
	const User = sequelize.define('user', {
		id: {
			allowNull: false,
			primaryKey: true,
			type: DataTypes.UUID,
			defaultValue: DataTypes.UUIDV4,
		},
		email: {
			type: DataTypes.STRING,
			validate: {
				isEmail: true,
			},
		},
		passwordHash: DataTypes.STRING,
		fullName: DataTypes.STRING,
		role: DataTypes.STRING,
	}, {});
	User.associate = (models) => {
		User.hasMany(models.meal);
		User.hasMany(models.menu);
		User.hasOne(models.profile);
	};
	return User;
};
