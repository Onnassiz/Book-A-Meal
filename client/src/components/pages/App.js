import React from 'react';
import AppContainer from '../container/AppContainer';

const App = () => (
  <div>
    <AppContainer />
  </div>
);

export default App;
