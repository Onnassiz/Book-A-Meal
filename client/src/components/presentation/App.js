import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import axios from 'axios';
import PropTypes from 'prop-types';
import Footer from '../presentation/layout/Footer';
import UserHomePage from '../pages/UserHomePage';
import Profile from '../pages/Profile';
import Home from '../pages/Home';
import Meals from '../pages/Meals';


class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			isSignedIn: false,
		};
	}

	componentWillMount() {
		axios.defaults.baseURL = process.env.API_URL;
		axios.defaults.headers.post['Content-Type'] = 'application/json; charset=utf-8';
		const token = localStorage.getItem('token');
		if (token !== '' && token !== null) {
			axios.defaults.headers.common.Authorization = `Bearer ${token}`;
			const { setUser } = this.props;
			const id = localStorage.getItem('id');
			const name = localStorage.getItem('name');
			const email = localStorage.getItem('email');
			const role = localStorage.getItem('role');
			const user = { id, name, email, role };

			setUser(user);
			this.setSignedIn();
		}
	}

	setSignedIn() {
		this.setState({ isSignedIn: true });
	}

	render() {
		const guestRoutes = (
			<Switch>
				<Route exact path="/" component={Home} />
			</Switch>
		);
		const userRoutes = (
			<Switch>
				<Route exact path="/" component={UserHomePage} />
				<Route exact path="/caterer/business_profile" component={Profile} />
				<Route exact path="/caterer/meals" component={Meals} />
			</Switch>
		);
		return (
			<div>
				<main>
					{ this.state.isSignedIn ? userRoutes : guestRoutes }
				</main>
				<Footer />
			</div>
		);
	}
}

App.propTypes = {
	setUser: PropTypes.func.isRequired,
};

export default App;
