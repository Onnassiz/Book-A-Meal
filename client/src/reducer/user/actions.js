import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actions';

export const SET_USER = 'SET_USER';
export const SET_SIGN_UP_SERVER_ERRORS = 'SET_SIGN_UP_SERVER_ERRORS';
export const SET_SIGN_IN_SERVER_ERRORS = 'SET_SIGN_IN_SERVER_ERRORS';

export function setUser(user) {
  return dispatch => dispatch({
    type: SET_USER,
    id: user.id,
    name: user.name,
    role: user.role,
    email: user.email,
  });
}

function setSignUpErrors(errors) {
  return dispatch => dispatch({
    type: SET_SIGN_UP_SERVER_ERRORS,
    errors,
  });
}

function setSignInErrors(errors) {
  return dispatch => dispatch({
    type: SET_SIGN_IN_SERVER_ERRORS,
    errors,
  });
}

export function postUser(user) {
  return (dispatch) => {
    dispatch(setLoading());
    axios.post('api/v1/auth/signUp', user).then((response) => {
      localStorage.setItem('id', response.data.id);
      localStorage.setItem('email', response.data.email);
      localStorage.setItem('name', response.data.fullName);
      localStorage.setItem('role', response.data.role);
      localStorage.setItem('token', response.data.token);
      setTimeout(() => {
        window.location = '/';
      }, 500);
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setSignUpErrors(error.response.data));
    });
  };
}


export function signInUser(user) {
  return (dispatch) => {
    dispatch(setLoading());
    axios.post('api/v1/auth/signIn', user).then((response) => {
      localStorage.setItem('id', response.data.id);
      localStorage.setItem('email', response.data.email);
      localStorage.setItem('name', response.data.fullName);
      localStorage.setItem('role', response.data.role);
      localStorage.setItem('token', response.data.token);
      setTimeout(() => {
        window.location = '/';
      }, 500);
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setSignInErrors(error.response.data));
    });
  };
}
