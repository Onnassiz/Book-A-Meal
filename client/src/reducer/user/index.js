import { SET_USER, SET_SIGN_UP_SERVER_ERRORS, SET_SIGN_IN_SERVER_ERRORS } from './action';

const initialState = {};

export default (state = initialState, action) => {
	switch (action.type) {
	case SET_USER:
		return { name: action.name, email: action.email, role: action.role };
	case SET_SIGN_UP_SERVER_ERRORS:
		return { signUpErrors: action.errors };
	case SET_SIGN_IN_SERVER_ERRORS:
		return { signInErrors: action.errors };
	default:
		return state;
	}
};
