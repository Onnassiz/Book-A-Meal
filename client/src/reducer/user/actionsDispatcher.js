import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actionsDispatcher';
import { setUserAction, setSignUpErrorsAction, setSignInErrorsAction } from './actions';

export const setUser = user => dispatch => dispatch(setUserAction(user));

const setSignUpErrors = errors => dispatch => dispatch(setSignUpErrorsAction(errors));
const setSignInErrors = errors => dispatch => dispatch(setSignInErrorsAction(errors));

const processSignIn = (data) => {
  localStorage.setItem('id', data.id);
  localStorage.setItem('email', data.email);
  localStorage.setItem('name', data.fullName);
  localStorage.setItem('role', data.role);
  localStorage.setItem('token', data.token);
  setTimeout(() => {
    window.location = '/';
  }, 500);
};

export const postUser = user => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/auth/signUp', user).then((response) => {
    dispatch(unsetLoading());
    processSignIn(response.data);
  }).catch((error) => {
    dispatch(setSignUpErrors(error.response.data));
    dispatch(unsetLoading());
  });
};


export const signInUser = user => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/auth/signIn', user).then((response) => {
    dispatch(unsetLoading());
    processSignIn(response.data);
  }).catch((error) => {
    dispatch(setSignInErrors(error.response.data));
    dispatch(unsetLoading());
  });
};
