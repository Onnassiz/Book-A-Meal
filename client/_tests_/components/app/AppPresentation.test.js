import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../../src/components/presentation/App';
import user from '../../objectProps/user.props';

const props = {
  setUser: jest.fn(),
  emptyCart: jest.fn(),
  cart: {},
  user,
};

const wrapper = shallow(<App {...props} />);

describe('<App />', () => {
  it('Should render App presentation components successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

