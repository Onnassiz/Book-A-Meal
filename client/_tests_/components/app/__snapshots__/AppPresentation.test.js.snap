// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render App presentation components successfully 1`] = `
<div>
  <main>
    <ToastContainer
      autoClose={3000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={true}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      pauseOnVisibilityChange={true}
      position="bottom-center"
      progressClassName={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/auth/:type"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
  <Component />
</div>
`;
